t/60-autoqc-checks-generic-artic.t

```diff
@@ -361,4 +361,33 @@ subtest 'saving JSON files' => sub {
     'error when the first mismatch is encountered';
 };
 
+subtest 'create correct list of result objects' => sub {
+  plan tests => 2;
+  
+  local $ENV{NPG_CACHED_SAMPLESHEET_FILE} =
+    't/data/autoqc/generic/artic/samplesheet_36214.csv';
+  # Tag metrics JSON file contains entries for both tag zero and 888.
+  # Tag 888 is registered in the samplesheet as a spiked phiX tag.
+  my $init = {
+    rpt_list => '36214:2',
+    pp_name  => 'artic',
+    pp_version => '0.10.0',
+    tm_json_file =>
+      't/data/autoqc/generic/artic/36214_2.with_spike.tag_metrics.json',
+    input_files_glob =>
+      't/data/autoqc/generic/artic/lane2/plex*/*.qc.csv'
+             };
+
+  my $check = npg_qc::autoqc::checks::generic::artic->new(%{$init});
+  my @results = @{$check->result()};
+  # Expected tag list does not contain either 0 or 888.
+  my @expected_tags =(1,2,3,4,97,137,140,157,159,160,205,206,207);
```

Comment:
I can see tag 0 in an earlier test. Is there scope for 888 appearing in another test? I'm not sure anything emits a 888 that hasn't been filtered by the new code.

Reply:
Line 339? The list is used to create output directories as they would be created by the pipeline. However, on line 348 we check that after running a full test there is no output for tag zero.

Reply:
What I mean is that the conditional filtering of the 888 tag does not seem to be tested anywhere. I'm not particularly concerned, but there's scope for a regression test.

Reply:
For over a year sequencing for this study was performed without spiked-in controls. Therefore, the samplesheet (the source of LIMS data for the analysis) used by the existing tests does not list the spiked-in phiX sample. Since this sample is not listed, the entry for its tag (usually indexed under 888) does not appear in the deplexing metrics (the `tm_json_file` attribute of the check object). The function I was changing creates entities for the tag indexes that are present in the deplexing metrics. The new test covers the scenario when both the samplesheet and the deplexing metrics do contain a spiked-in sample. A new lab procedure, which will result in this scenario, is about to start.

Reply:
Added explanation for the existing tests setup.

Reply:
Fair enough